From 620262b6b7b40cf160707c62cfbbd75d56ef51b1 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Fri, 06 May 2022 08:28:22 +0300
Subject: [PATCH] [SW-0] rom: update start address for iccm

As iccm is a closely coupled memory for arc, the performance
benefit can be utilized by executing code using internal addressing
and not by using cbu i/f addressing.

This patch converts the start address for arc1 iccm to internal
addressing.

Change-Id: Ida5b455200fd2864c2f43ad76cfbdfee4ae62304
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 5a6d64f..7b982be 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -28,13 +28,25 @@
 
 void arcboot_rom_bringup_arcboot_ram(void) {}
 
+uint32_t arcboot_rom_fw_mem_arc_addr(uint64_t addr)
+{
+	uint32_t ret_addr;
+
+	if (addr >= HL_ARC1_ICCM_FULL_BASE && addr <= HL_ARC1_ICCM_FULL_BASE + ARC1_ICCM_SIZE)
+		ret_addr = CONV_64B_TO_ARC_FW_MEM(addr) - (HL_FW_MEM_BASE + HL_ICCM_OFFSET);
+	else
+		ret_addr = CONV_64B_TO_ARC_FW_MEM(addr);
+
+	return ret_addr;
+}
+
 uint32_t arcboot_rom_get_arc_addr(uint64_t addr)
 {
 	if (addr >= HL_SRAM_BASE_FULL && addr <= HL_SRAM_BASE_FULL + HL_SRAM_SIZE)
 		return CONV_64B_TO_ARC_SRAM(addr);
 
 	if (addr >= HL_FW_MEM_BASE_FULL && addr <= HL_FW_MEM_BASE_FULL + HL_FW_MEM_SIZE)
-		return CONV_64B_TO_ARC_FW_MEM(addr);
+		return arcboot_rom_fw_mem_arc_addr(addr);
 
 	return 0;
 }
